src/Common/tests/Collections/ICollectionTest.cs

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Linq;
+using Xunit;
+using Xunit.Sdk;
+// ReSharper disable InconsistentNaming
+
+namespace Tests.Collections
+{
+    public abstract class ICollectionTest<T> : IEnumerableTest<T>
+    {
+        internal class MyInvalidReferenceType
+        {
+        }
+
+        internal struct MyInvalidValueType
+        {
+        }
+
+        protected ICollection GetCollection(object[] items)
+        {
+            IEnumerable obj = GetEnumerable(items);
```

Comment:
This assumption is in the ICollectionTest class. If someone makes an implementation of ICollectionTest<T> and the collection doesn't implement ICollection, that is not allowed.